### PR TITLE
chart.js added types to generateLabels function

### PR DESCRIPTION
**Pull request by wertzui**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.chartjs.org/docs/latest/configuration/legend.html#legend-label-configuration
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/chart.js/index.d.ts [hunk 1]
```diff
@@ -270,7 +271,7 @@ declare namespace Chart {
         fontColor?: ChartColor;
         fontFamily?: string;
         padding?: number;
-        generateLabels?(chart: any): any;
+        generateLabels?(chart: ChartData): ChartLegendLabelItem[];
```
Comment (adripanico) [inline types/chart.js/index.d.ts]: According to the documentation, it should be generateLabels?(chart: Chart): ChartLegendItem[];

**Review by adripanico — CHANGES_REQUESTED** [review]

File: types/chart.js/index.d.ts (hunk 1 above)
Comment (wertzui) [reply types/chart.js/index.d.ts]: Thanks, I changed it to `Chart`

Comment (adripanico) [reply types/chart.js/index.d.ts]: Also ChartLegendLabelItem[] it's supposed to be ChartLegendItem[].

Comment (wertzui) [reply types/chart.js/index.d.ts]: No, it has the `datasetIndex` property and therefore is a `ChartLegendLabelItem`

Comment (adripanico) [reply types/chart.js/index.d.ts]: In that snippet they are attaching such property but the element is defined without it: https://www.chartjs.org/docs/latest/configuration/legend.html#legend-item-interface.

Actually, the link you shared is clearly identifying datasetIndex as an extra parameter.

Comment (wertzui) [reply types/chart.js/index.d.ts]: Then the definition of `onClick` must be updated too to reflect that change.

Comment (wertzui) [reply types/chart.js/index.d.ts]: @adripanico would you rather see the `onClick` method definition changed, or keep the `ChartLegendLabelItem`?

Comment (adripanico) [reply types/chart.js/index.d.ts]: It's ok. I'll approve it.

**Review by adripanico — APPROVED** [review]

**Pull request merged**